Wikipedia/Code/MWKDataStore.m

```diff
@@ -495,6 +529,53 @@ - (void)iterateOverArticles:(void (^)(MWKArticle*))block {
     }
 }
 
+- (void)startCacheRemoval {
+    dispatch_async(self.cacheRemovalQueue, ^{
+        self.cacheRemovalActive = true;
```

Comment:
`cacheRemovalActive` would still be necessary in the NSOp case - to check whether or not the op should be added to the queue. you could nil out the queue, but that would prevent necessary operations from going through - namely writing the work list to disk.